### ISSUE TITLE
Add JSR 305 annotations.

### ISSUE THREAD

**Issue by JakeWharton** [square/javapoet#565]
Lots of platform types when writing annotation processors in Kotlin which makes me 😢

**Comment by JakeWharton** [comment]
Changing title as JSR 305 is toxic as of Java 9.

**Comment by tbroyer** [comment]
Or the checker framework? https://youtrack.jetbrains.com/issue/KT-21982

**Comment by JakeWharton** [comment]
Yeah I mean there's tons that Kotlin recognizes: https://github.com/JetBrains/kotlin/blob/39f4a449ad43bf1a5496e96c9d0e48b0c342e764/core/descriptors.jvm/src/org/jetbrains/kotlin/load/java/JvmAnnotationNames.kt#L21-L34

I was hoping [KEEP-99][1] would move quickly which would let us specify a package-global default similar to JSR 305.

 [1]: https://github.com/Kotlin/KEEP/issues/99
